### PR TITLE
[updates] add doc comments in source

### PR DESCRIPTION
**Pull request by Simek**
# Why

Fixes ENG-1625

To utilize the auto generation feature for API section on the docs website the package source code must contain the relevant documentation.

# How

This PR moves the current documentation of `Updates` API - https://docs.expo.io/versions/latest/sdk/updates - with minor tweaks and additions to the package source code.

To full support the migration to automated docs extraction I had to introduce few improvements and fixes to the code:
* `UpdateEvent`, `UpdateCheckResult` and `UpdateFetchResult` types generalisation
* listener event type and `LocalAssets` simplification
* hide in docs manifest related types because they do not explain anything about the structure of manifest right now, as the comment in code states:
  > modern manifest type is intentionally not exported, since the plan is to call it just "Manifest" in the future 
* hide in docs experimental (and probably deprecated) `clearUpdateCacheExperimentalAsync` method
* do not link `EventSubscription` type from `fbemitter`

There are entries which have not been earlier exposed in the docs, so now they are missing the documentation.

I'm not sure if any of the types changes qualifies for the changelog, so at this moment I have leave it black, but I can add few entries if you think they are required. 🙂 

# Test Plan

The changes and generated content have been tests running docs website on `localhost`.

# Preview

Screenshot 2021-07-27 at 15 19 17

# Checklist

- [X] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).
- [X] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [X] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Review by bbarthec — APPROVED** [review]
From my end it looks fine, but let's wait for @esamelson review in terms of changing the exported elements 🖕

File: packages/expo-updates/src/Updates.ts [hunk 1]
```diff
@@ -18,19 +17,54 @@ import {
...
     ? ExpoUpdates.updateId.toLowerCase()
     : null;
+
+// @needsAudit
+/**
+ * The name of the release channel currently configured in this standalone or bare app.
+ */
 export const releaseChannel: string = ExpoUpdates.releaseChannel ?? 'default';
+
+// @docsMissing
```
Comment (esamelson) [inline packages/expo-updates/src/Updates.ts:38]:
```suggestion
// @docsMissing
/**
 * @hidden
 */
```

File: packages/expo-updates/src/Updates.ts [hunk 2]
```diff
@@ -18,19 +17,54 @@ import {
...
+ * you don't need to worry about backwards compatibility when you put out an update. In very rare
+ * cases, it's possible that `expo-updates` may need to fall back to the update that's embedded in
+ * the app binary, even after newer updates have been downloaded and run (an "emergency launch").
+ * This boolean will be `true` if the app is launching under this fallback mechanism and `false`
+ * otherwise. If you are concerned about backwards compatibility of future updates to your app, you
+ * can use this constant to provide special behavior for this rare case.
+ */
 export const isEmergencyLaunch: boolean = ExpoUpdates.isEmergencyLaunch || false;
+
+// @docsMissing
```
Comment (esamelson) [inline packages/expo-updates/src/Updates.ts:53]:
```suggestion
// @docsMissing
/**
 * @hidden
 */
```

File: packages/expo-updates/src/Updates.types.ts [hunk 3]
```diff
@@ -1,25 +1,81 @@
...
+   */
+  manifest?: Manifest;
+  /**
+   * If `type === Updates.UpdateEventType.ERROR`, the error message. Undefined otherwise.
+   */
+  message?: string;
+};
 
-export type LocalAssets = { [remoteUrl: string]: string };
+// @docsMissing
```
Comment (esamelson) [inline packages/expo-updates/src/Updates.types.ts:80]:
```suggestion
// @docsMissing
/**
 * @hidden
 */
```

**Review by esamelson — APPROVED** [review]
The `localAssets` and `isUsingEmbeddedAssets` fields are only used internally (by other modules) and we don't want to document them for now. Otherwise LGTM! 😄

**Pull request merged**